core/src/main/java/org/elasticsearch/search/suggest/SuggestUtils.java

```diff
@@ -208,78 +202,6 @@ public static StringDistance resolveDistance(String distanceVal) {
-            if (parseFieldMatcher.match(fieldName, Fields.ACCURACY)) {
-                suggestion.accuracy(parser.floatValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.SUGGEST_MODE)) {
-                suggestion.suggestMode(SuggestUtils.resolveSuggestMode(parser.text()));
-            } else if (parseFieldMatcher.match(fieldName, Fields.SORT)) {
-                suggestion.sort(SortBy.resolve(parser.text()));
-            } else if (parseFieldMatcher.match(fieldName, Fields.STRING_DISTANCE)) {
-                suggestion.stringDistance(SuggestUtils.resolveDistance(parser.text()));
-            } else if (parseFieldMatcher.match(fieldName, Fields.MAX_EDITS)) {
-                suggestion.maxEdits(parser.intValue());
-                if (suggestion.maxEdits() < 1 || suggestion.maxEdits() > LevenshteinAutomata.MAXIMUM_SUPPORTED_DISTANCE) {
-                    throw new IllegalArgumentException("Illegal max_edits value " + suggestion.maxEdits());
-                }
-            } else if (parseFieldMatcher.match(fieldName, Fields.MAX_INSPECTIONS)) {
-                suggestion.maxInspections(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MAX_TERM_FREQ)) {
-                suggestion.maxTermFreq(parser.floatValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.PREFIX_LENGTH)) {
-                suggestion.prefixLength(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MIN_WORD_LENGTH)) {
-                suggestion.minWordLength(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MIN_DOC_FREQ)) {
-                suggestion.minDocFreq(parser.floatValue());
-            } else {
-                return false;
-            }
-            return true;
-    }
-
-    public static boolean parseSuggestContext(XContentParser parser, MapperService mapperService, String fieldName,
-            SuggestionSearchContext.SuggestionContext suggestion, ParseFieldMatcher parseFieldMatcher) throws IOException {
-
-        if (parseFieldMatcher.match(fieldName, Fields.ANALYZER)) {
-            String analyzerName = parser.text();
-            Analyzer analyzer = mapperService.analysisService().analyzer(analyzerName);
-            if (analyzer == null) {
-                throw new IllegalArgumentException("Analyzer [" + analyzerName + "] doesn't exists");
-            }
-            suggestion.setAnalyzer(analyzer);
-        } else if (parseFieldMatcher.match(fieldName, Fields.FIELD)) {
-            suggestion.setField(parser.text());
-        } else if (parseFieldMatcher.match(fieldName, Fields.SIZE)) {
-            suggestion.setSize(parser.intValue());
-        } else if (parseFieldMatcher.match(fieldName, Fields.SHARD_SIZE)) {
-            suggestion.setShardSize(parser.intValue());
-        } else {
-           return false;
-        }
-        return true;
-    }
```

Comment:
Parsing this fields has moved to each individual SuggestionBuilders `innerFromXContent` method, so IMHO good to go.